src/client/content_cao.cpp

```diff
@@ -1502,8 +1502,9 @@ void GenericCAO::processMessage(const std::string &data)
 		// localplayer itself can't be attached to localplayer
 		if (!m_is_local_player) {
 			m_attached_to_local = getParent() != NULL && getParent()->isLocalPlayer();
-			// Objects attached to the local player should be hidden by default
-			m_is_visible = !m_attached_to_local;
+			// Objects attached to the local player should be hidden in first person
+			m_is_visible = (!m_attached_to_local) 
+				|| (m_client->getCamera()->getCameraMode() > CAMERA_MODE_FIRST);
```

Comment:
why > ? it's error prone if we add more things in the enum fields. I prefer != if possible

Reply:
I suppose it could have been useful if more camera modes were added where the entities shouldn't be shown (in which case, they would be set below CAMERA_MODE_FIRST), but that seems unlikely, in retrospect, so I've changed it.